src/Compilers/CSharp/Test/Semantic/Semantics/RecordTests.cs

```diff
@@ -28708,5 +28592,343 @@ public void SealedIncomplete()
+  .maxstack  2
+  IL_0000:  ldarg.1
+  IL_0001:  ldarg.0
+  IL_0002:  ldfld      ""int A.X""
+  IL_0007:  stind.i4
+  IL_0008:  ldarg.2
+  IL_0009:  ldarg.0
+  IL_000a:  ldfld      ""int A.Y""
+  IL_000f:  stind.i4
+  IL_0010:  ret
+}
+");
+        }
+
+        [Fact]
+        public void FieldAsPositionalMember_Readonly()
+        {
+            var source = @"
+var a = new A(42);
+System.Console.Write(a.X);
+System.Console.Write("" - "");
+a.Deconstruct(out int x);
+System.Console.Write(x);
+
+record A(int X)
+{
+    public readonly int X = X;
+}
+";
+            var comp = CreateCompilation(source, parseOptions: TestOptions.RegularPreview);
+            comp.VerifyDiagnostics();
+            var verifier = CompileAndVerify(comp, expectedOutput: "42 - 42", verify: Verification.Skipped /* init-only */);
+            verifier.VerifyIL("A.Deconstruct", @"
+{
+  // Code size        9 (0x9)
+  .maxstack  2
+  IL_0000:  ldarg.1
+  IL_0001:  ldarg.0
+  IL_0002:  ldfld      ""int A.X""
+  IL_0007:  stind.i4
+  IL_0008:  ret
+}
+");
+        }
+
+        [Fact]
+        public void FieldAsPositionalMember_UnusedParameter()
+        {
+            var source = @"
+record A(int X)
+{
+    public int X;
+}
+";
+            var comp = CreateCompilation(source, parseOptions: TestOptions.RegularPreview);
+            comp.VerifyEmitDiagnostics(
+                // (2,14): warning CS8907: Parameter 'X' is unread. Did you forget to use it to initialize the property with that name?
+                // record A(int X)
+                Diagnostic(ErrorCode.WRN_UnreadRecordParameter, "X").WithArguments("X").WithLocation(2, 14),
+                // (4,16): warning CS0649: Field 'A.X' is never assigned to, and will always have its default value 0
+                //     public int X;
+                Diagnostic(ErrorCode.WRN_UnassignedInternalField, "X").WithArguments("A.X", "0").WithLocation(4, 16)
+                );
+        }
+
+        [Fact]
+        public void FieldAsPositionalMember_PropertyComesFirst()
+        {
+            var source = @"
+var c = new C(0);
+c.Deconstruct(out int i);
+System.Console.Write(i);
+
+public record Base
+{
+    public int I { get; set; } = 42;
+    public Base(int ignored) { }
+}
+public record C(int I) : Base(I)
+{
+    public int I;
+}
+";
+            var comp = CreateCompilation(source, parseOptions: TestOptions.RegularPreview);
+            comp.VerifyDiagnostics(
+                // (13,16): warning CS0108: 'C.I' hides inherited member 'Base.I'. Use the new keyword if hiding was intended.
+                //     public int I;
+                Diagnostic(ErrorCode.WRN_NewRequired, "I").WithArguments("C.I", "Base.I").WithLocation(13, 16)
+                );
+
+            var verifier = CompileAndVerify(comp, expectedOutput: "42");
+            verifier.VerifyIL("C.Deconstruct", @"
+{
+  // Code size        9 (0x9)
+  .maxstack  2
+  IL_0000:  ldarg.1
+  IL_0001:  ldarg.0
+  IL_0002:  call       ""int Base.I.get""
```

Comment:
>IL_0002:  call       ""int Base.I.get"" [](start = 2, length = 39)

This looks unexpected. I think `C.I` should be used here.

Reply:
And that is brobably going to be a breaking change.

---
In reply to: [612824515](https://github.com/dotnet/roslyn/pull/52480#discussion_r612824515) [](ancestors = 612824515)

Reply:
It looks like you intentionally tried to preserve previous behavior, but I am not sure if the proposed behavior is the right on to have and whether we explicitly approved it at LDM. 

---
In reply to: [612824890](https://github.com/dotnet/roslyn/pull/52480#discussion_r612824890) [](ancestors = 612824890,612824515)

Reply:
Added an open LDM issue to test plan.
[https://github.com/dotnet/roslyn/issues/51199](https://github.com/dotnet/roslyn/issues/51199)

I'll start email thread and also put record structs on the agenda for next Monday, since multiple small questions remain.

---
In reply to: [612828289](https://github.com/dotnet/roslyn/pull/52480#discussion_r612828289) [](ancestors = 612828289,612824890,612824515)

Reply:
> Added an open LDM issue to test plan.

Untill this is settled, I do not think we should merge this sub-feature.

---
In reply to: [612941755](https://github.com/dotnet/roslyn/pull/52480#discussion_r612941755) [](ancestors = 612941755,612828289,612824890,612824515)